src/main/java/org/cirdles/topsoil/app/TopsoilMainWindow.java

```diff
@@ -310,29 +354,60 @@ void initializeAndShow(Chart chart, Dataset dataset) {
     }
 
     private void initializeAndShow(JavaScriptChart javaScriptChart) {
-        getCurrentTable().map(TSVTable::getDataset).ifPresent(dataset -> {
-            initializeAndShow(javaScriptChart, dataset);
+        getCurrentTable().ifPresent(table -> {
+            initializeAndShow(javaScriptChart, table.getDataset());
         });
     }
 
     @FXML
     void createScatterplot() {
-        initializeAndShow(new ScatterplotChart());
+        if (getCurrentTable().isPresent()) {
+                initializeAndShow(new ScatterplotChart());
+        } else {
+            //Not a TSVTable
+            warningPrompt("You can't draw a chart from this table.\nPlease select a valid table first.");
+        }
     }
 
     @FXML
     void createErrorEllipseChart() {
-        initializeAndShow(new ErrorEllipseChart());
+        if (getCurrentTable().isPresent()) {
+                initializeAndShow(new ErrorEllipseChart());
+        } else {
+            //Not a TSVTable
+            warningPrompt("You can't draw a chart from this table.\nPlease select a valid table first.");
+        }
     }
 
     @FXML
     void pasteFromClipboard() {
-        getCurrentTable().ifPresent(TSVTable::pasteFromClipboard);
+        getCurrentTable()
+                .ifPresent(
+                        currentTable -> currentTable.pasteFromClipboard()
+                );
     }
 
     @FXML
     void emptyTable() {
-        getCurrentTable().ifPresent(TSVTable::clear);
+        if (getCurrentTable().isPresent()) {
+            getCurrentTable().get().clear();
+        } else {
+            //Not a TSVTable
+            warningPrompt("You can't clear this table.");
```

Comment:
And if it's the instructions panel, it warns "can't clear table", hence the confusion.